duckduckgo/duckduckgo-privacy-extension
# Add domains to whitelist from settings page

**Reviewer:** @andrey-p 
**CC:** @alohaas 



## Description:
Introduce the ability to add new domains to the whitelist from settings.
<img width="459" alt="screen shot 2018-02-21 at 1 43 24 am" src="https://user-images.githubusercontent.com/3652195/36466527-abbb4744-16a8-11e8-824c-119099f66bd1.png">
<img width="486" alt="screen shot 2018-02-21 at 1 43 17 am" src="https://user-images.githubusercontent.com/3652195/36466528-abd23e72-16a8-11e8-8e23-03f055ddbac5.png">
<img width="474" alt="screen shot 2018-02-21 at 1 43 02 am" src="https://user-images.githubusercontent.com/3652195/36466529-abddf758-16a8-11e8-9de4-83a9c5d1b2b1.png">

## Steps to test this PR:
1. Build and reload extension
2. Go to settings page
3. "Add site to whitelist" link-button should be visible, input and button should be hidden
4. Click on "Add site to whitelist". Page does not jump, and the link should then be hidden, while the input and button should be visible now
5. Input should be empty and button should be disabled and have the default cursor on hover
6. When typing in the input, the button should be enabled (blue with white font) and cursor on hover should be the pointer
7. Removing all the input should disable the button again
8. Either clicking on the button or hitting return inside the input should submit the domain
9. If the domain is not valid, an error popout should appear and the input box border should turn red. That all disappears when the user types again in the input box
10. If the domain is valid, it should be added to the whitelist, and in this case too the page will now look like at step 3
11. Clicking on any of the whitelisted domain's remove buttons should remove the respective domain from the list
12. Navigate to the whitelisted sites to make sure they're actually whitelisted

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
